Standardise component structure (#150)
sa-tre/satre-specification: docs/source/contributing/walkthrough.md

Please fill in [this form](https://forms.office.com/pages/responsepage.aspx?id=p_SVQ1XklU-Knx-672OE-XMfskt9FY5HraQ6BiwaEGBUNUg1T05DNEJMNEtaN0JOOURKTDBFUENFQi4u) to register your attendance in advance.

Collaboration Cafés are run using HackMD documents that will contain the meeting agenda and a place to share notes.
If you are unfamiliar with HackMD, please see [this guide](https://hackmd.io/@turingway/hackmd-guide) on how to use it.
Please also see a [sample HackMD](https://hackmd.io/N7EBtM6iS2OQsWCPWC-ojg) from our Collaboration Café on the 18th May 2023.
Each Collaboration Café uses Breakout Rooms where participants will work on themes, e.g. Risk Management or Training Requirements.
Typically, each Breakout Room is linked to an Issue on the [SATRE GitHub Issues Board](https://github.com/sa-tre/satre-specification/issues).

:::{important}
The HackMD for each Collaboration Café is made available in your calendar invites.
We encourage you to propose a Breakout Room in advanced of the Collaboration Café, linking to a [SATRE GitHub Issue](https://github.com/sa-tre/satre-specification/issues) if possible.

> This will help participants come prepared to talk about certain topics. We also welcome Breakout Room proposals on the day, though!

Author: ```suggestion
This will help participants come prepared to talk about certain topics. We also welcome Breakout Room proposals on the day!
```

> This will help participants come prepared to talk about certain topics. We also welcome Breakout Room proposals on the day, though!

Author: ```suggestion
This will help participants come prepared to talk about certain topics. We also welcome Breakout Room proposals on the day!
```

There are two ways to do this:

1. Directly to the specification on GitHub (recommended, but more difficult)
2. To a non-GitHub version of the specification (less recommended, but easier)

Why the difference?
GitHub is the live version of the specification, and is great for things like community discussions, version control, editing changes and more.
This is why we recommend contributing via GitHub - it is the most direct way to work with the SATRE team on the specification.

However, we realise this may be difficult if you aren't already familiar with GitHub's ways of working.
Therefore, we also have a non-GitHub version of the specification you can comment on.
Once a week, we will translate comments from this version to the GitHub repository for direct amendments.
So you are still able to contribute via this method, though all direct changes will still be taking place on GitHub.

### Contribute directly through GitHub

Below is a walkthrough of all the steps required to contribute via GitHub, from the very beginning.

This walkthrough is designed to give you the critical path steps to contributing directly to the Specification repository.

> For a more broad, deeper introduction to GitHub, check out the Turing Way's [Introduction to GitHub Workshop](https://www.youtube.com/watch?v=Vcckl-2dASM), run at CarpentryCon22.

Reviewer: ```suggestion
For a broader, deeper introduction to GitHub, check out the Turing Way's [Introduction to GitHub Workshop](https://www.youtube.com/watch?v=Vcckl-2dASM), run at [CarpentryCon2022](https://2022.carpentrycon.org/).
```

1. Once you are logged in, navigate to the [SATRE Specification page](https://github.com/sa-tre/satre-specification)

2. This is the SATRE specification **repository**. You can think of this like a directory containing all files to do with the SATRE specification.

>    The main things to know about are the link in the Section, and the `Code`, `Issues` and `Pull requests` tabs.

Reviewer: Not sure why you deleted `About`? Isn't it it referring to the link to the website in the `About` section?

Author: Auto-formatter error. It interprets `text` as whitespace and so wants to replace " `About` " with " ". Good catch.

>    The main things to know about are the link in the Section, and the `Code`, `Issues` and `Pull requests` tabs.

Author: ```suggestion
   The main things to know about are the link in the `About` Section, and the `Code`, `Issues` and `Pull requests` tabs.
```

Pull request number
```

> There will always be a dedicated Breakout Room in the Collaboration Cafés where one of the SATRE Team will be on hand to answer any questions and guide you through the GitHub Repository and help you get set up.

Reviewer: ```suggestion
There will always be a dedicated Breakout Room in the Collaboration Cafés where one of the SATRE Team will be on hand to answer any questions, guide you through the GitHub Repository, and help you get set up.
```
